modules/compactor/compactor.go

```diff
@@ -45,50 +56,94 @@ func New(cfg Config, store storage.Store, overrides *overrides.Overrides, reg pr
 		overrides: overrides,
 	}
 
-	subservices := []services.Service(nil)
 	if c.isSharded() {
-		lifecyclerCfg := c.cfg.ShardingRing.ToLifecyclerConfig()
-		lifecycler, err := ring.NewLifecycler(lifecyclerCfg, ring.NewNoopFlushTransferer(), "compactor", cfg.OverrideRingKey, false, log.Logger, prometheus.WrapRegistererWithPrefix("cortex_", reg))
+		reg = prometheus.WrapRegistererWithPrefix("cortex_", reg)
+
+		lifecyclerStore, err := kv.NewClient(
+			cfg.ShardingRing.KVStore,
+			ring.GetCodec(),
```

Comment:
This will match up with the codec we are currently using? i.e. on rollout all components will still gossip successfully?

Reply:
Yes this is a copy of the behavior that was in [NewLifecycler()](https://github.com/grafana/tempo/blob/v1.2.1/vendor/github.com/cortexproject/cortex/pkg/ring/lifecycler.go#L155) that is no longer in `NewBasicLifecycler`.